wallet/src/repository/wallet/mod.rs

```diff
@@ -267,32 +307,33 @@ where
                 index,
             },
         )?;
+
+        let next_index = index.checked_add(1).ok_or_else(|| Error::IndexOverflow)?;
         batch.put(keys::account_next_index(account, keychain), &next_index)?;
 
         self.db.write(batch)?;
 
-        let address = model::Address {
-            address,
-            path,
-            info,
-            index,
-            account,
-            keychain,
-            pkh,
-        };
+        // Return previous address index (always 1 address ahead)
+        let address = self._get_address(state, account, keychain, index).ok();
```

Comment:
This ignores any errors returned my `_get_address`. If `address` should be `None` in some special cases, handle that case specially. For example, if `_get_address` returns `None` when the index is 0:

```suggestion
        let address = if index != 0 { Some(self._get_address(state, account, keychain, index)?) } else { None };
```

Reply:
Or even better, make `_get_address` return an `Option`